examples/ui-demo/src/state/store.tsx

```diff
@@ -0,0 +1,209 @@
+    typeof window === "undefined"
+      ? createInitialState(initialConfig)
+      : persist(createInitialState(initialConfig), {
+          name: STATE_KEY,
+          storage: createJSONStorage(() => cookieStorage()),
+          partialize: ({
+            setAuth,
+            setBorderRadius,
+            setIllustrationStyle,
+            setLogo,
+            setPrimaryColor,
+            setSupportUrl,
+            setTheme,
+            setNftTransferred,
+            nftTransferred,
+            ...config
+          }) => config,
+          skipHydration: true,
+          version: 1,
+        })
+  );
+};
+
+function createInitialState(
+  initialConfig: Config = DEFAULT_CONFIG
+): StateCreator<DemoState> {
+  return (set, get) => ({
+    ...initialConfig,
+    nftTransferred: false,
+    setNftTransferred: (transferred) => {
+      set({ nftTransferred: transferred });
+    },
+    setTheme: (theme) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          theme,
+        },
+      }));
+    },
+    setSupportUrl: (url) => {
+      set(() => ({
+        supportUrl: url,
+      }));
+    },
+    setLogo: (theme, logo) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          [theme]: logo,
+        },
+      }));
+    },
+    setPrimaryColor: (theme, color) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          primaryColor: {
+            ...state.ui.primaryColor,
+            [theme]: color,
+          },
+        },
+      }));
+    },
+    setAuth: (auth) => {
+      set((state) => ({
+        auth: {
+          ...state.auth,
+          ...auth,
+        },
+      }));
+    },
+    setIllustrationStyle: (style: Config["ui"]["illustrationStyle"]) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          illustrationStyle: style,
+        },
+      }));
+    },
+    setBorderRadius: (radius: Config["ui"]["borderRadius"]) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          borderRadius: radius,
+        },
+      }));
+    },
+  });
+}
+
+function deserialize<type>(value: string): type {
+  return JSON.parse(decodeURIComponent(value));
+}
+
+export function cookieToInitialConfig(cookie?: string | null) {
+  if (!cookie) return;
+
+  const state = parseCookie(cookie, STATE_KEY);
+  if (!state) return;
+
+  const config = deserialize<{ state: Config }>(state).state;
+
+  return config;
+}
+
+export function generateStylesForRoot(config: Config) {
+  const primaryColor = config.ui.primaryColor[config.ui.theme];
+
+  return [
+    [getColorVariableName("fg-accent-brand"), primaryColor],
+    [getColorVariableName("btn-primary"), primaryColor],
+    [
+      getBorderRadiusBaseVariableName(),
+      getBorderRadiusValue(config.ui.borderRadius),
+    ],
+  ];
+}
+
+export function generateClassesForRoot(config: Config): string[] {
+  return [config.ui.theme];
+}
+
+export function deepEqual(a: any, b: any): boolean {
```

Comment:
because the demo app's config store isn't super deep or complex, this ensures that fewer renders take place when nested state is updated. the cost of the deep equal is sub 1ms after doing some testing